src/core/log_callback.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include <string>
+#include <functional>
+
+namespace mavsdk::log {
+
+// defined numeric values can be useful for filtering message by severity
+typedef enum {
+  Debug = 0,
+  Info = 1,
+  Warn = 2,
+  Err = 3
+} Level;
+
+/** @brief User-defined callback for logging. Returning true from this callback
+ * prevents default mavsdk`s logging to stdout. Returning false keeps it.
+ */
+typedef std::function<bool(Level level, const std::string &message,
```

Comment:
I'd prefer to use the more modern:

```
using Callback = std::function<bool(Level level, const std::string& message, const std::string& file, int line)>;`
```

Reply:
Thanks, will fix. Also apparently some styling checks failed. I'll wait a bit for discussion here to settle and then fix everything at once